processor/processorhelper/traces.go

```diff
@@ -36,14 +36,26 @@ type tracesProcessor struct {
 	consumer.Traces
 }
 
-// NewTracesProcessor creates a TracesProcessor that ensure context propagation and the right tags are set.
-// TODO: Add observability metrics support
+// Deprecated: [v0.58.0] use version with NewTracesProcessorWithCreateSettings.
 func NewTracesProcessor(
 	cfg config.Processor,
 	nextConsumer consumer.Traces,
 	tracesFunc ProcessTracesFunc,
 	options ...Option,
 ) (component.TracesProcessor, error) {
+	return NewTracesProcessorWithCreateSettings(context.Background(), component.ProcessorCreateSettings{}, cfg, nextConsumer, tracesFunc, options...)
```

Comment:
Are you planning to bring back `NewTracesProcessor` name once the old one is removed?

Reply:
yes. we need intermediate to keep backwards compatibility for 2 versions.